### PR TITLE
add bare environment CI tests

### LINKED ISSUE #727: Add CI test for bare environment installation
#725 describes how the CI builds let an installation dependency issue slip through to a new release. We should improve the Azure Pipelines config to prevent that from happening.

Might make sense to do this along with removing the `2.7-min` build next month.

### PR DESCRIPTION
**Pull request by wholmgren**
- [x] Closes #727
 - [x] I am familiar with the [contributing guidelines](http://pvlib-python.readthedocs.io/en/latest/contributing.html).
 - [x] Fully tested. Added and/or modified tests to ensure correct behavior for all reasonable inputs. Tests (usually) must pass on the TravisCI and Appveyor testing services.
 - [x] Updates entries to `docs/sphinx/source/api.rst` for API changes.
 - [x] Adds description and name entries in the appropriate `docs/sphinx/source/whatsnew` file for all changes.
 - [x] Code quality and style is sufficient. Passes LGTM and SticklerCI checks.
 - [x] New code is fully documented. Includes sphinx/numpydoc compliant docstrings and comments in the code where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -6,6 +6,43 @@ trigger:
...
+    matrix:
+      Python35:
+        python.version: '3.5'
+      Python36:
+        python.version: '3.6'
+      Python37:
+        python.version: '3.7'
+
+  steps:
+  - task: UsePythonVersion@0
```
Comment (markcampanelli) [inline azure-pipelines.yml]: Just for my own information, where does the choice of @ versions come from here and below?

Comment (wholmgren) [reply azure-pipelines.yml]: I'm not sure that I understand the scope of the question, so let me know if this doesn't actually answer it. It's probably fine to just choose 1 python version for this bare environment test, but it's easy enough to add the matrix option so I figured why not. In general, we want to test pvlib against all supported Python versions. This is stated in the readme and installation docs as "3.5 and above". 3.8 just came out, so I will look into adding that in a few weeks. I'm pretty confident that pvlib itself will not have any issues with 3.8, but it often takes a few weeks for the dependencies to release new wheels and/or Anaconda to add compiled packages.

**Comment by wholmgren** [comment]
Ready to merge. Test failures are due to `get_psm3`.

File: azure-pipelines.yml (hunk 1 above)
Comment (markcampanelli) [reply azure-pipelines.yml]: More specifically, how do you know what the @0, @1, and @3 suffixes refer to?

Comment (wholmgren) [reply azure-pipelines.yml]: Oh, I don’t understand that part of the spec. I’m just following the docs here https://docs.microsoft.com/en-us/azure/devops/pipelines/ecosystems/python?view=azure-devops

Comment (wholmgren) [reply azure-pipelines.yml]: Specifies the task version number: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/tasks?view=azure-devops&tabs=yaml#task-versions

File: docs/sphinx/source/whatsnew/v0.7.0.rst [hunk 2]
```diff
@@ -129,6 +129,8 @@ Testing
   in NSRDB (:issue:`733`)
 * Added tests for methods in bifacial.py.
 * Added tests for changes to cell temperature models.
+* Add tests configuration for bare python environment (no conda).
```
Comment (cwhanse) [inline docs/sphinx/source/whatsnew/v0.7.0.rst]: I can't comment on the test configuration being added. But here: past tense "Added" or present tense "Add"?  No reason to hold up the PR.

Comment (wholmgren) [reply docs/sphinx/source/whatsnew/v0.7.0.rst]: There's a mix of past and present throughout the file, so I'm going to keep it as is for now. I'm ok setting a loose standard if someone wants to do the background research into what the rest of the python ecosystem uses. I can see arguments for either past or present. Separate issue.

Comment (cwhanse) [reply docs/sphinx/source/whatsnew/v0.7.0.rst]: [Python](https://docs.python.org/3/whatsnew/changelog.html) release notes mix past and present tense, as does [numpy](https://docs.scipy.org/doc/numpy/release.html). Not worth the trouble to create our own convention and enforce it.

File: azure-pipelines.yml (hunk 1 above)
Comment (markcampanelli) [reply azure-pipelines.yml]: Oh I think I get it now. Microsoft supplies those task versions so they don’t break people’s pipelines when they make changes.

**Pull request merged**